src/gui/folderstatusmodel.cpp

```diff
@@ -712,9 +712,11 @@ void FolderStatusModel::slotUpdateDirectories(const QStringList &list)
         newSubs.append(newInfo);
     }
 
-    beginInsertRows(idx, 0, newSubs.size() - 1);
-    parentInfo->_subs = std::move(newSubs);
-    endInsertRows();
+    if (!newSubs.isEmpty()) {
+        beginInsertRows(idx, 0, newSubs.size() - 1);
+        parentInfo->_subs = std::move(newSubs);
+        endInsertRows();
+    }
```

Comment:
parentInfo->_subs should be empty.  I added an assert now to verify it.
Otherwise, the insertRows would also be wrong as it would be replacing rows.